File: leetcode-solutions/go/118. Pascal's Triangle.go
// https://leetcode.com/problems/pascals-triangle/

func generate(numRows int) [][]int {
	// rows := make([]int, numRows)
	result := make([][]int, numRows)
	for i := 0; i < numRows; i++ {
			result[i] = make([]int, i+1)
	}
	for i := 0; i < numRows; i++ {
			for j := 0; j <= i; j++ {
					if j == 0 || j == i {
							result[i][j] = 1
					}else {
							result[i][j] = result[i-1][j-1] + result[i-1][j]
					}
			} 
	}   
	
	return result
}

File: leetcode-solutions/python/118. Pascal's Triangle.py
# https://leetcode.com/problems/pascals-triangle/

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = []
        for row in range(1, numRows + 1):
            triangle.append([])
            for col in range(row):
                if col == 0 or col == row - 1:
                    triangle[-1].append(1)
                else:
                    triangle[-1].append(triangle[row - 2][col] + triangle[row - 2][col - 1])
        return triangle